mri/operators/fourier/non_cartesian.py

```diff
@@ -378,9 +385,127 @@ def adj_op(self, x):
         return img * np.sqrt(np.prod(self.Kd))
 
 
+class gpuNUFFT:
+    """  GPU implementation of N-D non uniform Fast Fourrier Transform class.
+
+    Attributes
+    ----------
+    samples: np.ndarray
+        the mask samples in the Fourier domain.
+    shape: tuple of int
+        shape of the image (necessarly a square/cubic matrix).
+    operator: The NUFFTOp object
+        to carry out operation
+    n_coils: int default 1
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition. If n_coils > 1, please organize data as
+            n_coils X data_per_coil
+    """
+    def __init__(self, samples, shape, n_coils=1, density_comp=None,
+                 kernel_width=3, sector_width=8, osf=2, balance_workload=True,
+                 smaps=None):
+        """ Initilize the 'NUFFT' class.
+
+        Parameters
+        ----------
+        samples: np.ndarray
+            the mask samples in the Fourier domain.
+        shape: tuple of int
+            shape of the image
+        n_coils: int
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition
+        density_comp: np.ndarray default None.
+            k-space weighting, density compensation, if not specified
+            equal weightage is given.
+        kernel_width: int default 3
+            interpolation kernel width (usually 3 to 7)
+        sector_width: int default 8
+            sector width to use
+        osf: int default 2
+            oversampling factor (usually between 1 and 2)
+        balance_workload: bool default True
+            whether the workloads need to be balanced
+        smaps: np.ndarray default None
+            Holds the sensitivity maps for SENSE reconstruction
+        """
+        if (n_coils < 1) or (type(n_coils) is not int):
+            raise ValueError('The number of coils should be an integer >= 1')
+        self.n_coils = n_coils
+        self.shape = shape
+        if samples.min() < -0.5 or samples.max() >= 0.5:
+            warnings.warn("Samples will be normalized between [-0.5; 0.5[")
+            samples = normalize_frequency_locations(samples)
+            self.samples = samples
+        if density_comp is None:
+            density_comp = np.ones(samples.shape[0])
+        if smaps is not None:
+            self.uses_smaps = False
+        else:
+            self.uses_smaps = True
+        self.operator = NUFFTOp(
```

Comment:
I guess this will raise a `NotFoundError`. Maybe it would be better when doing the import to have a flag set. This way we can give a better error message here.

Reply:
Makes sense, will add that

Reply:
you didn't use the flag at the beginning of the call to `__init__`